open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java

```diff
@@ -1865,6 +1865,142 @@ public void   updateSchemaAttribute(String               userId,
+                    OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_GUID,
+                    OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_NAME,
+                    forLineage,
+                    forDuplicateProcessing,
+                    supportedZones,
+                    instanceProperties,
+                    true,
+                    effectiveTime,
+                    methodName);
+
+            SchemaTypeBuilder schemaTypeBuilder = schemaAttributeBuilder.getSchemaTypeBuilder();
+            if (schemaTypeBuilder != null)
+            {
+                // todo this logic assumes the schema type is stored as a classification
+                setClassificationInRepository(userId,
+                                              externalSourceGUID,
+                                                      externalSourceName,
+                                                      schemaAttributeEntity,
+                                                      schemaAttributeGUIDParameterName,
+                                                      attributeTypeName,
+                                                      OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_GUID,
+                                                      OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_NAME,
+                                                      schemaTypeBuilder.getTypeEmbeddedInstanceProperties(methodName),
+                                                      isMergeUpdate,
+                                                      forLineage,
+                                                      forDuplicateProcessing,
+                                                      supportedZones,
+                                                      effectiveTime,
+                                                      methodName);
+            }
+            /*
+             * The formula is set if the schema attribute is derived
+             */
+            if (formula != null && schemaTypeBuilder != null)
+            {
+                schemaAttributeBuilder.setCalculatedValue(userId, externalSourceGUID, externalSourceName, formula, methodName);
+
+                setClassificationInRepository(userId,
+                                              externalSourceGUID,
+                                              externalSourceName,
+                                              schemaAttributeEntity,
+                                              schemaAttributeGUIDParameterName,
+                                              attributeTypeName,
+                                              OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_GUID,
+                                              OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_NAME,
+                                              schemaTypeBuilder.getTypeEmbeddedInstanceProperties(methodName),
```

Comment:
To set formula, this method should be calling schemaAttributeBuilder.getCalculatedValueProperties(), and so the test for null schemaTypeBuilder is erroneus.  The classification should be set if isMergeUpdate is true or if formula is not null.  To be consistent, the formula should probably be passed in schemaAttributeBuilder?  But then need to work out how to properly supportthe isMergeUpdate semantics.

I suspect it is wrong in the other method since you say you do not understand its intent?

Reply:
Hi Mandy, I have pushed up a change, as the formula is in the schema type - I can expose it in the schema type builder and mot have to pass it down as a parameter.  It is still a bit odd
- the formula is in the schematype (and builder)
- the calculated values are in the attribute builder

I have updated the code, but want to test more and review on Monday.

Reply:
@mandy-chessell I have amended my testing to create a formula using the lineage context. On the createSchemaAttribute The formula  comes  though in the on the schematype , but is not converted into the calculatedValue classification. The entity is only created with the embedded type and anchors classifications. I am looking into this - this fix will be extended to cover the create case.

Reply:
<img width="698" alt="Screenshot 2022-12-05 at 15 31 53" src="https://user-images.githubusercontent.com/39792797/205676878-aa88b711-9789-4e59-a2e5-51c355a971a5.png">

Reply:
I have made some changes to allow the create schema attribute to honour a supplied formula. I have tested create and update .